src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -1394,17 +1421,59 @@ export async function update(
     return { status: 'error' }
   }
 
-  //install the new version
-  const installResult = await install(appName, {
-    path: gameInfo.install.install_path,
-    platformToInstall: gameInfo.install.platform,
-    channelName: gameInfo.install.channelName,
-    accessCode: args?.accessCode,
-    updateOnly: true,
-    siweValues: args?.siweValues
-  })
-
+  // try patching first, if it fails, download the new version
   const isMarketWars = gameInfo.account_name === 'marketwars'
+
+  const {
+    channels,
+    install: { channelName, platform, install_size, install_path, executable }
+  } = gameInfo
+
+  if (
+    !channels ||
+    !channelName ||
+    !platform ||
+    !install_path ||
+    !executable ||
+    install_size === undefined
+  ) {
+    logError(
+      `Channel name or platform not found for ${appName} in update`,
+      LogPrefix.HyperPlay
+    )
+    throw new Error('Channel name or platform not found')
+  }
+
+  const newVersion = channels[channelName].release_meta.name
+  const abortController = createAbortController(appName)
+  const { status, error } = await applyPatching(
+    gameInfo,
+    newVersion,
+    abortController.signal
+  )
+
+  if (status === 'abort') {
+    return { status: 'abort' }
+  } else if (status === 'error' && !error?.includes('aborted')) {
+    // if error, download the zip file
+    const installResult = await install(appName, {
+      path: gameInfo.install.install_path,
+      platformToInstall: gameInfo.install.platform,
+      channelName: gameInfo.install.channelName,
+      accessCode: args?.accessCode,
+      updateOnly: true,
+      siweValues: args?.siweValues
+    })
+    if (isMarketWars) {
+      try {
+        await runModPatcher(appName)
+      } catch (error) {
+        return { status: 'error' }
+      }
```

Comment:
this type of code snippet is a result of us mixing JS errors and our own status string and having to convert between the two. 

In the future, I think we could simplify this considerably if we threw a custom error object with a status property. here's an example of how we could do that https://github.com/HyperPlay-Gaming/hyperplay-dev/blob/main/utils/EndpointError.ts